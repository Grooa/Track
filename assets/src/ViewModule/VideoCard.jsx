import PropTypes from 'prop-types';

import CommonCard from '../common/CommonCard';

const VideoCard = ({ video, number }) => {
  const type = `Video ${number}`;
  const description = video.shortDescription || '';
  const hasAccess = !!video.url;
  const button = hasAccess ? 'Checkout' : 'You don\'t have access';

  return <CommonCard
    type={type}
    title={video.title}
    link={hasAccess ? video.url : null}
    description={description}
    buttonLabel={button}
    isDisabled={!hasAccess} />;
};

VideoCard.propTypes = {
  video: PropTypes.object.isRequired,
  number: PropTypes.number.isRequired,
};

export default VideoCard;
