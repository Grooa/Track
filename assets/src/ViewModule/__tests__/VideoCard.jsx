import { render } from 'enzyme';

import VideoCard from '../VideoCard';

describe('<VideoCard />', () => {
  let video;

  beforeEach(() => {
    video = {
      title: 'Test title',
      shortDescription: 'hello world',
      url: 'https://test.com/test',
    };
  });

  it('should render properly', () => {
    const wrapper = render(<VideoCard video={video} number={1}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render disabled properly', () => {
    video.url = null;

    const wrapper = render(<VideoCard video={video} number={1}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
