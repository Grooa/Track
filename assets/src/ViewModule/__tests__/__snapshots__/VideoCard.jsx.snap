// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoCard /> should render disabled properly 1`] = `
<div
  class="course-module inactive"
>
  <div
    class="course-module-metadata"
  >
    <strong
      class="type"
    >
      Video 1
    </strong>
    <h3
      class="title"
    >
      Test title
    </h3>
  </div>
  <div
    class="description"
  >
    <p>
      hello world
    </p>
  </div>
  <button>
    You don't have access
  </button>
</div>
`;

exports[`<VideoCard /> should render properly 1`] = `
<a
  class="course-module clickable"
  href="https://test.com/test"
>
  <div
    class="course-module-metadata"
  >
    <strong
      class="type"
    >
      Video 1
    </strong>
    <h3
      class="title"
    >
      Test title
    </h3>
  </div>
  <div
    class="description"
  >
    <p>
      hello world
    </p>
  </div>
  <button>
    Checkout
  </button>
</a>
`;
